### ISSUE TITLE
assign mouse buttons to zooming in automap

### ISSUE THREAD

**Issue by fabiangreffrath** [chocolate-doom/chocolate-doom#608]
I just had a request to allow for assigning mouse buttons (i.e. the mouse wheel) to zooming in and out in the automap. I general, I think this feature would fit into Chocolate Doom as well, but I am afraid that it will clash with weapon changing which is usually applied to the buttons.

Except for a few exceptions, button presses are not eaten in the automap but passed through to the actual game. To support this feature properly we would have to eat these buttons in the automap, but I guess this would break demos in which a weapon has been changed while the automap was active. What do you think?

**Comment by fabiangreffrath** [comment]
No, wait. Only the act of changing weapons is saved in the demo file, not the actual key/button press. So, this could still work, or not?

**Comment by chungy** [comment]
I believe it can work. Doesn't the next/previous weapon just simulate pressing the next/previous number?

**Comment by fabiangreffrath** [comment]
Incidentally, it appears that SDL2 changes the way the mouse wheel is handled. So, maybe we should postpone this request until after we have finished porting to SDL2.

**Comment by haleyjd** [comment]
Yeah, mouse wheel events (up or down) work mostly like expected in SDL2 rather than being exceptional events that misbehave in terms of giving you an up or down state. Handling them there is quite natural.

**Comment by jmtd** [comment]
We can revisit this now mouse wheel events are handled in sdl2-branch
